.github/workflows/build_and_run_esp_usb_test_apps.yml

```diff
@@ -0,0 +1,80 @@
+name: ESP-USB Test Apps
+
+on:
+  schedule:
+    - cron: '0 0 * * *' # Every day at midnight
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      fail-fast: false
+      matrix:
+        idf_ver: ["release-v5.0", "release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "latest"]
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    env:
+      ESP_USB_MANIFEST: ./esp-usb/.build-test-rules.yml
+      ESP_TINYUSB_TEST_APPS: ./esp-usb/device/esp_tinyusb/test_apps
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - name: Clone esp-usb repository
+        run: |
+            git clone https://github.com/espressif/esp-usb.git
+      - name: Build
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install idf-component-manager==1.5.2 idf-build-apps==2.4.3 --upgrade
+          export PEDANTIC_FLAGS="-DIDF_CI_BUILD -Werror -Werror=deprecated-declarations -Werror=unused-variable -Werror=unused-but-set-variable -Werror=unused-function"
+          export EXTRA_CFLAGS="${PEDANTIC_FLAGS} -Wstrict-prototypes"
+          export EXTRA_CXXFLAGS="${PEDANTIC_FLAGS}"
+          python .github/ci/override_managed_component.py tinyusb . ${{ env.ESP_TINYUSB_TEST_APPS }}/*/
+          idf-build-apps find --path ${{ env.ESP_TINYUSB_TEST_APPS }}/ --recursive --target esp32s2 --manifest-file ${{ env.ESP_USB_MANIFEST }}
```

Comment:
But I don't see the reason, why I need build it here, as I won't use the artifacts later. 
Possibility, that there is some differences in the code between S2 and S3 are zero, as the code for dcd_dwc2 for them are equal. 
The tinyusb basically provides basically only device stack (classes + dcd).

Also, there is a complication of keeping the artifacts, because the recently added lines, as  `--build-dir build_@t_@w` squashes the directory path, and I still have no idea how to resolve it correctly when we need the same directory path during two different steps: build and run, when we use the external repo to get the examples source code.

Reply:
Just a suggestion to also build it for esp32s3. We do it in esp-usb, as well as in esp-idf (to build the code for all supported targets) as a sanity check.

In esp-usb, we also just upload all the artifacts, even if we don't use them later in the run job.

Yes if you decide to do it, you will have to also modify the code at couple more places. 

Also, then once you will be adding build and run for esp32p4 (which I think you will) you will have to modify the code in a very similar way you would do now, to add the esp32s3 build.

To deal with the build directories for different targets, you may want to take a look at the `/esp-usb/.idf_build_apps.toml` file

Reply:
Ok, agreed. 
I will add esp32s3 and esp32p4 in the next following MRs.